src/utils/resources/template/hooks/useVmTemplateSource/utils.ts

```diff
@@ -108,6 +108,19 @@ export const getDataSourcePVC = (name: string, ns: string) =>
 export const isDataSourceReady = (dataSource: V1beta1DataSource): boolean =>
   dataSource?.status?.conditions?.some((c) => c.type === 'Ready' && c.status === 'True');
 
+/**
+ * a function that returns true if the data source is cloning in progress
+ * @param dataSource the data source to check if cloning
+ * @returns true if the data source is in cloning state, false otherwise
+ */
+export const isDataSourceCloning = (dataSource: V1beta1DataSource): boolean =>
+  dataSource?.status?.conditions?.some(
+    (c) =>
+      c.type === 'Ready' &&
```

Comment:
Do the conditions always have `type`, `status` and `reason` present?

Reply:
`reason` is optional. I'll fix that. Thanks for the good catch